zpay32/invoice.go

```diff
@@ -288,24 +288,17 @@ func Decode(invoice string) (*Invoice, error) {
 	}
 
 	// The next characters should be a valid prefix for a segwit BIP173
-	// address. This will also determine which network this invoice is
-	// meant for.
-	var net *chaincfg.Params
-	if strings.HasPrefix(hrp[2:], chaincfg.MainNetParams.Bech32HRPSegwit) {
-		net = &chaincfg.MainNetParams
-	} else if strings.HasPrefix(hrp[2:], chaincfg.TestNet3Params.Bech32HRPSegwit) {
-		net = &chaincfg.TestNet3Params
-	} else if strings.HasPrefix(hrp[2:], chaincfg.SimNetParams.Bech32HRPSegwit) {
-		net = &chaincfg.SimNetParams
-	} else {
+	// address that match the active network.
```

Comment:
I guess the `if len(hrp) < 4 ` test above is not strictly correct anymore (if the segwit prefix is 1 character). Instead we should check the length as we go (check at least two characters before doing "ln" check, at least 1 char before doing segwit prefix check), or just check it is a least 3 :P

Reply:
Yeah I'm thinking only a length check of < 3 up front makes the most sense as that is the shortest possible valid HRP. Length check afterwards seems like an if statement that will never resolve to true, because if we got past the network check it means we matched the `Bech32HRPSegwit` prefix exactly - length and all.